Create the charter for Kubeflow Community Distribution committee (#951)
kubeflow/community: KUBEFLOW_DISTRIBUTION_COMMITTEE.md

### Out of Scope (Limitations)

- Anti-Ecosystem-Competitive Actions: Engaging in or permitting actions that undermine the longevity of the community or violate vendor-neutral sovereignty. This includes, but is not limited to:
- **Proprietary Lock-in:** Mandating that sub-projects or the distribution depend on proprietary APIs or closed-source components when general, open-source alternatives exist.
- **Core Fragmentation:** Implementing core functionality that is not upstreamed, as the KDC’s goal is to maintain a Kubernetes-native, portable, and extensible foundation.
- **Exclusionary Integration:** Forcing dependencies that intentionally disadvantage specific vendors or environments.
- **Note on Compatibility:** This does not exclude the development of community-supported or vendor-contributed adapters (e.g., data exporters, storage drivers, or external dependencies) designed to support specific environments, provided they remain external to the project's core functionality and do not compromise the portability of the base ecosystem.

## Committee Structure

> ### Committee Roles

Reviewer: ```suggestion
### Committee Roles

Currently, all representatives are appointed by a normal decision process of the KSC.
```

> Changes to the KDC charter can be proposed by any community member via a GitHub PR. Amendments will be subject to approval by a standard decision of the KSC. Proposals will be available for at least one week for community comments before a vote occurs.

Reviewer: Can you move this to the end of the document like here, please?
https://github.com/kubeflow/community/blob/3f8c0027b1690bbf16d47b63816e9904e8c0d7c0/KUBEFLOW-OUTREACH-COMMITTEE.md#changes-to-the-charter

Author: Done

## Committee Structure

### Committee Roles

1. 40% - Votes by Appointment

>     - 3-4 seats appointed by KSC for 1 year term

Reviewer: @thesuperzapper @juliusvonkohout @franciscojavierarceo @chasecadet Do we want to allow same person be in the vendor sit and KCD sit?

Author: lets hope that we first get enough real participants. My guess is that this can often be the case, because maintainers tend to do it for a company in some way. So my answer is yes I do not want to make it illegal. If we see problems the KSC can restrict it later,

    - 30% - KCD maintainers - 1-2 seats, all votes are weighted in total by 30%
    - 30% - Every sub-project gets 1 seat, all votes are weighted in total by 30%
    - Is allowed to vote on Kubeflow Community Distribution (KCD) topics
    - When voting decisions happen on KCD topics, maintainers and sub-project owners get 50% / 50% voting power

>   

Reviewer: ```suggestion

### Decision Process

####  Normal Decision Process

Decisions requiring a vote include:

- Any decisions on requirements for Kubeflow conformance program
- Any decisions on requirements for Kubeflow subprojects portability and ease of deployment
- Any other decisions that at least half of the chairs (rounded down) present decide requires a vote

All representatives are allowed to vote on these topics. 

Decisions are made with pull requests to the community repository, must be documented in the `committee-distribution/decision_log.md`, and vote publicly on GitHub. Votes may only pass when at least 50% of the binding vote weight have been cast.

The decisions is adopted if more then half of the votes (rounded up) support it, taking into account the weighting of those votes.

#### KCD Decision Process

Some issues related to KCD requires a special decision process. Issues include:

- Any decisions on requirements for KCD
- Propose projects for KSC to be included in KCD. List of projects that are included in KCD must be defined in  `committee-distribution/PROJECTS.md`

Only KCD and subprojects representatives are allowed to vote on these topics.

Decisions are made with pull requests to the community repository, must be documented in the `committee-distribution/decision_log_kcd.md`, and vote publicly on GitHub. Votes may only pass when at least 50% of the binding vote weight have been cast.

The decisions is adopted if more then half of the votes (rounded up) support it, taking into account the weighting of those votes.
```

### Committee members

> KCD maintainers:

Reviewer: ```suggestion
KCD representatives:
```

| Name                      | Organization    | GitHub                                                     |
| ------------------------- | --------------- | ---------------------------------------------------------- |
| Julius von Kohout         | DHL Data & AI   | [@juliusvonkohout](https://github.com/julusvonkohout/)     |
| Tarek Abouzeid            | Telia           | [tarekabouzeid](https://github.com/tarekabouzeid/)         |

>           

Reviewer: ```suggestion

Kubeflow Subproject Representatives

| Name                      | Subproject    | GitHub                                                     |
| ------------------------- | --------------- | ---------------------------------------------------------- |
| TODO         | Kubeflow Trainer  | [@TODO](https://github.com/TODO/)     |

```

Vendor represantatives:
| Name                      | Organization    | GitHub                                                     |
| ------------------------- | --------------- | ---------------------------------------------------------- |
|                           |                 |                                                            |

> The subprojects votes are not bound to individual members.

Reviewer: ```suggestion
```

| ------------------------- | --------------- | ---------------------------------------------------------- |

> | Tarek Abouzeid            | Telia           | [tarekabouzeid](https://github.com/tarekabouzeid/)         |

Reviewer: Let's onboard vendors separately? Once we merge the charter, we can call out for any vendors who are interested to be in the committee.

Author: Done

The subprojects votes are not bound to individual members.

### Limitations on Company Representation
No more than one maintainer or vendor seat may be held by employees of the same organization (or conglomerate, in the case of companies owning each other). 

If employers change because of job changes, acquisitions, or other events, in a way that would be in violation of the proceeding limits, sufficient members of the committee must resign their positions until the requirements are satisfied. If it is impossible to find sufficient members to resign, all employees of that organization will be removed and the vacancies will be filled using the normal process.

In the event of a question of company membership (for example evaluating independence of corporate subsidiaries) a majority of all non-involved KSC members will decide.

### Committee Meetings

The KDC will meet regularly in the existing "Kubeflow AI Reference Platform (Release, Manifests, Security)" meeting which will be renamed to "Kubeflow Community Distribution (Release, Manifests, Security)". These meetings will be open to the public. Specific sensitive matters may be handled separately.

> Meeting notes will be publicly available to the community, except when privacy is required for specific discussions.

Reviewer: I don't think we need to keep historical info in the charter.

```suggestion
The KDC will meet regularly during Kubeflow Community Distribution Call (Release, Manifests, Security). These meetings will be open to the public. Specific sensitive matters may be handled separately.

Meeting notes will be publicly available to the community, except when privacy is required for specific discussions.
```

If employers change because of job changes, acquisitions, or other events, in a way that would be in violation of the proceeding limits, sufficient members of the committee must resign their positions until the requirements are satisfied. If it is impossible to find sufficient members to resign, all employees of that organization will be removed and the vacancies will be filled using the normal process.

In the event of a question of company membership (for example evaluating independence of corporate subsidiaries) a majority of all non-involved KSC members will decide.

### Committee Meetings

The KDC will meet regularly in the existing "Kubeflow AI Reference Platform (Release, Manifests, Security)" meeting which will be renamed to "Kubeflow Community Distribution (Release, Manifests, Security)". These meetings will be open to the public. Specific sensitive matters may be handled separately.

Meeting notes will be publicly available to the community, except when privacy is required for specific discussions.

## Kubeflow community distribution maintainers

### In Scope (minimal expectations)

> - Synchronize the application and dependencies manifests to then elaborately combine (configure) them for a consistent, secure and end-to-end multi-tenant enterprise platform experience.

Reviewer: ```suggestion
- The Kubeflow Community Distribution must be Kubeflow Conformant. 
- Synchronize the application and dependencies manifests to then elaborately combine (configure) them for a consistent, secure and end-to-end multi-tenant enterprise platform experience.
```

- Maintain the evolving and not exhaustive list of included dependencies for a proper multi-tenant platform installation: Istio, KNative, Dex, Oauth2-proxy, Cert-Manager, ... as well as Kubeflow sub-projects, Kubeflow Ecosystem projects and disabled by default experimental integrations.
- Release tested releases of the Kubeflow Community Distribution for downstream consumption.
- Provide guidance and examples how the consumer could integrate non-default external authentication (e.g., companies' Identity Provider) and popular non-default services on his own.
- Make Kubeflow Community Distribution compatible with the popular Kubernetes clusters (Kind, Rancher, AKS, GKE, EKS, OpenShift)
- Aid the kubeflow subproject maintainer in creating manifests (Helm, Kustomize) for his application.
- Aid the kubeflow subproject maintainer regarding security best practices.

> - Communicate with the kubeflow subproject maintainer regarding releases and versioning.

Reviewer: Can we combine this?

```suggestion
- Aid the Kubeflow subproject maintainers in creating manifests (Helm, Kustomize), security best practices, maintain releases and versioning for their application.
```

> # Kubeflow Community Distribution

Reviewer 2: We need to add a section that makes it clear that the KDC charter (as in, this document) is only changed by a vote of the KSC: 

https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md#changes-to-the-charter

The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable.

> ## Charter

Reviewer 2: We should consider if we want to enforce that this comittee meets in a public meeting at least monthly, similar to the KOC:

https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md#committee-meetings

Reviewer: Agree, we can re-use Kubeflow Release team for that too. cc @tarekabouzeid @varodrig

Author: Yes, just the same call.

### Out of Scope (Limitations)

- Anti-Ecosystem-Competitive Actions: Engaging in or permitting actions that undermine the longevity of the community or violate vendor-neutral sovereignty. This includes, but is not limited to:
      - **Proprietary Lock-in:** Mandating that sub-projects or the distribution depend on proprietary APIs or closed-source components when general, open-source alternatives exist.
       - **Core Fragmentation:** Implementing core functionality that is not upstreamed, as the KDC’s goal is to maintain a Kubernetes-native, portable, and extensible foundation.
       - **Exclusionary Integration:** Forcing dependencies that intentionally disadvantage specific vendors or environments.
       - **Note on Compatibility:** This does not exclude the development of community-supported or vendor-contributed adapters (e.g., data exporters, storage drivers, or external dependencies) designed to support specific environments, provided they remain external to the project's core functionality and do not compromise the portability of the base ecosystem.

> ## Committee Structure

Reviewer 2: We need to create a new section called "Comitte Members" which lists current members and their terms (or the fact there are vacant seats which need to be appointed by each WG).

See KOC for example: https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md#committee-members-1

### Committee Roles

> 1. 40% - Votes by Appointment

Reviewer 2: In the interest of making sure this somewhat complex voting system is enforced when decisoins are raised into the `kubeflow/community` repo, we should configure gitvote (which is what the CNCF TOC uses) to ensure the correct conditions are met for a change to be adopted by the KDC.

https://github.com/cncf/gitvote

Reviewer: I agree, could you open separate issue to track it please, @thesuperzapper?

    - 3-4 seats appointed by KSC for 1 year term
    - Only vendor representative.
    - Is NOT allowed to vote on Kubeflow Community Distribution (KCD) topics

2. 60% - Votes by Right
    - 30% - KCD maintainers - 1-2 seats, all votes are weighted in total by 30%
    - 30% - Every sub-project gets 1 seat, all votes are weighted in total by 30%
    - Is allowed to vote on Kubeflow Community Distribution (KCD) topics
    - When voting decisions happen on KCD topics, maintainers and sub-project owners get 50% / 50% voting power

## Kubeflow community distribution maintainers

> 

Reviewer 2: If we plan to allow the KDC to vote to change this section:

we should simply make it the first "decision" of the KDC, and __move it under the `kubeflow/community` folder where we will catalog decisions of the KDC__ in its own markdown file called `decision_1.md`

Reviewer: Good point, @thesuperzapper maybe we could add a folder `committee-distribution`, where we are going to have sub-folders?
E.g.
```
commitee-distribution/
  README.md          # Members are defined here + some useful info
  charter.md              # We will move main charter here.
  decision-log.md.    # Changes to conformance, security, etc.
  kubeflow-community-distribution/
    README.md
    OWNERS
    charter.md           # Short overview for the KCD 
    decision-log.md  # Changes to the KCD
```

WDYT about it? We can do similar changes to KSC / KOC if we like it.

I noticed k8s has similar structure: https://github.com/kubernetes/community/tree/main/committee-security-response

Thoughts @juliusvonkohout @franciscojavierarceo @chasecadet @thesuperzapper ?

Author: I prefer to have as few files as possible and everything in one file.

Reviewer: I guess, the goal of such directory is to allow Community Distribution chairs approve and vote on changes without KSC approval.

If employers change because of job changes, acquisitions, or other events, in a way that would be in violation of the proceeding limits, sufficient members of the committee must resign their positions until the requirements are satisfied. If it is impossible to find sufficient members to resign, all employees of that organization will be removed and the vacancies will be filled using the normal process.

In the event of a question of company membership (for example evaluating independence of corporate subsidiaries) a majority of all non-involved KSC members will decide.

### Committee Meetings

The KDC will meet regularly in the existing "Kubeflow AI Reference Platform (Release, Manifests, Security)" meeting which will be renamed to "Kubeflow Community Distribution (Release, Manifests, Security)". These meetings will be open to the public. Specific sensitive matters may be handled separately.

Meeting notes will be publicly available to the community, except when privacy is required for specific discussions.

## Kubeflow community distribution maintainers

> ### In Scope (Powers)

Reviewer: ```suggestion
### Goals
```

Author: done

- Synchronize the application and dependencies manifests to then elaborately combine (configure) them for a consistent, secure and end-to-end multi-tenant enterprise platform experience.

> - Enable the consumer and distributions to install, extend, modify and maintain Kubeflow installations by providing documentation, automation and configurability that covers (but is not limited to) installation, extension, security, and architecture decisions in order to enable the consumer to run a machine learning platform on Kubernetes. He may also choose to install individual components without the benefits of the platform. Everyone is free to create his own distribution for any number of customers or companies in private or in public by deriving from Kubeflow Community Distribution or building from scratch.

Reviewer: ```suggestion
- Enable the consumer and distributions to install, extend, modify and maintain Kubeflow Community Distribution installations by providing documentation, automation and configurability that covers (but is not limited to) installation, extension, security, and architecture decisions in order to enable the consumer to run a machine learning platform on Kubernetes. He may also choose to install individual components without the benefits of the platform. Everyone is free to create his own distribution for any number of customers or companies in private or in public by deriving from Kubeflow Community Distribution or building from scratch.
```

- Maintain an extensive testing suite in order to cover scenarios that the community member and consumer expects from an AI/ML platform. This includes dependencies, security efforts, and exemplary integration with popular tools and frameworks and making sure via integration tests that the components work end-to-end together as a secure and multi-tenant platform. 

> - Maintain the evolving and not exhaustive list of included dependencies for a proper multi-tenant platform installation: Istio, KNative, Dex, Oauth2-proxy, Cert-Manager, ... as well as integrating the Kubeflow applications: Kubeflow Pipelines, Trainer, Dashboard, Workspaces / Notebooks, KServe, Spark, ...

Reviewer: ```suggestion
- Maintain the evolving and not exhaustive list of included dependencies for a proper multi-tenant platform installation: Kubeflow sub-projects, Kubeflow Ecosystem projects, and integration projects to enable Kubeflow Community Integration. List of projects that are included in Kubeflow Community Distribution are listed below in the charter.
```

- Release tested releases of the Kubeflow Community Distribution for downstream consumption.
- Provide guidance and examples how the consumer could integrate non-default external authentication (e.g., companies' Identity Provider) and popular non-default services on his own.

> - Try to be compatible with the popular Kubernetes clusters (Kind, Rancher, AKS, GKE, EKS, OpenShift, ...).

Reviewer: ```suggestion
- Make Kubeflow Community Distribution cloud agnostic and be compatible with the popular Kubernetes clusters (Kind, Rancher, AKS, GKE, EKS, OpenShift, etc.)
```

> - Try to be compatible with the popular Kubernetes clusters (Kind, Rancher, AKS, GKE, EKS, OpenShift, ...).

Reviewer: ```suggestion
- Try to be compatible with the popular Kubernetes clusters (Kind, Rancher, AKS, GKE, EKS, OpenShift, ...).
- Collaborate with Kubeflow maintainers to aid in creation of manifests (Helm, Kustomize), security best practices, maintain releases, and version compatibility.
```

> ### Collaboration with maintainers of kubeflow projects (e.g. KFP, Trainer, ...)

Reviewer: Let's move this to goals.

- Aid the kubeflow project maintainer in creating manifests (Helm, Kustomize) for his application.
- Aid the kubeflow project maintainer regarding security best practices.

> - Communicate with the kubeflow project maintainer regarding releases and versioning.

Reviewer: ```suggestion
```

### Out of Scope (Limitations)

- Anti-Ecosystem-Competitive Actions: Engaging in or permitting actions that undermine the longevity of the community or violate vendor-neutral sovereignty. This includes, but is not limited to:
      - **Proprietary Lock-in:** Mandating that sub-projects or the distribution depend on proprietary APIs or closed-source components when general, open-source alternatives exist.
       - **Core Fragmentation:** Implementing core functionality that is not upstreamed, as the KDC’s goal is to maintain a Kubernetes-native, portable, and extensible foundation.
       - **Exclusionary Integration:** Forcing dependencies that intentionally disadvantage specific vendors or environments.
       - **Note on Compatibility:** This does not exclude the development of community-supported or vendor-contributed adapters (e.g., data exporters, storage drivers, or external dependencies) designed to support specific environments, provided they remain external to the project's core functionality and do not compromise the portability of the base ecosystem.

### Changes to the Charter

> Changes to the KOC charter can be proposed by any community member via a GitHub PR. Amendments will be subject to approval by a standard decision of the KSC. Proposals will be available for at least one week for community comments before a vote occurs.

Reviewer 3: ```suggestion
Changes to the KDC charter can be proposed by any community member via a GitHub PR. Amendments will be subject to approval by a standard decision of the KSC. Proposals will be available for at least one week for community comments before a vote occurs.
```

# Kubeflow Community Distribution

> The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable. The charter must only be changed by the KSC.

Reviewer: Let's use the same section as in KOC doc: https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md#changes-to-the-charter
```
Changes to the Charter
Changes to the KOC charter can be proposed by any community member via a GitHub PR. Amendments will be subject to approval by a standard decision of the KSC. Proposals will be available for at least one week for community comments before a vote occurs.
```

```suggestion
The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable.
```

# Kubeflow Community Distribution committee charter

> The Kubeflow Kubeflow Community Distribution is a committee dedicated to provide a tested and integrated platform experience for the Kubeflow community.

Reviewer 4: ```suggestion
The Kubeflow Community Distribution is a committee dedicated to provide a tested and integrated platform experience for the Kubeflow community.
```

Reviewer 5: +1

> # Kubeflow Community Distribution committee charter

Reviewer: ```suggestion
# Kubeflow Community Distribution

The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable.

```

Reviewer: @juliusvonkohout I've updated the description based on @chasecadet suggestion.

- Ensure that the consumer can leverage and modify our integrated platform in order to create an opinionated distribution that satisfies his individual requirements. He may also choose to install individual components without the benefits of the platform. Everyone is free to create his own distribution for any number of customers or companies in private or in public by deriving from Kubeflow Community Distribution or building from scratch
- Release tested releases of the Kubeflow Community Distribution for downstream consumption.
- Try to be compatible with the popular Kubernetes clusters (Kind, Rancher, Azure Kubernetes Service, Elastic Kubernetes Service, Google Kubernetes Engine, ...).
- Provide hints and experimental examples how the consumer could integrate non-default external authentication (e.g., companies' Identity Provider) and popular non-default services on his own.
- Document the installation of Kubeflow as a platform and/or individual components including common problems and architectural overviews.

### Out of scope

- We do not enforce a specific deployment tool (e.g., ArgoCD, Flux), but we also do not want to block the usage of ArgoCD.

> - The default installation shall not contain deep integration with external cloud services or closed source solutions; instead, we aim for Kubernetes-native solutions and light authentication and authorization integration with external identity providers.

Reviewer: Based on @chasecadet suggestion in the doc.

```suggestion
### Out of Scope (Limitations)

- Anti-Ecosystem-Competitive Actions: Engaging in or permitting actions that undermine the longevity of the community or violate vendor-neutral sovereignty. This includes, but is not limited to:
      - **Proprietary Lock-in:** Mandating that sub-projects or the distribution depend on proprietary APIs or closed-source components when general, open-source alternatives exist.
       - **Core Fragmentation:** Implementing core functionality that is not upstreamed, as the KDC’s goal is to maintain a Kubernetes-native, portable, and extensible foundation.
       - **Exclusionary Integration:** Forcing dependencies that intentionally disadvantage specific vendors or environments.
       - **Note on Compatibility:** This does not exclude the development of community-supported or vendor-contributed adapters (e.g., data exporters, storage drivers, or external dependencies) designed to support specific environments, provided they remain external to the project's core functionality and do not compromise the portability of the base ecosystem.

```

# Kubeflow Community Distribution committee charter

- The Kubeflow Community Distribution is a committee dedicated to provide a tested and integrated platform experience for the Kubeflow community.
The governance of the Kubeflow Community Distribution committee will evolve as the community grows, ensuring that the platform remains relevant and impactful.

> - Promote the vision, values, and mission of the Kubeflow project in collaboration with the KSC, KOC, and WG Leads.

Reviewer: This should be part of the Charter, like for KOC and KSC: https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md

Author: Done

- Enable all Kubeflow sub-projects to be integrated and deployed as a cohesive, end-to-end platform.
- Improve the ease of deployment, packaging, and distribution of Kubeflow sub-projects across heterogenous environments.
- Empower external organizations to adopt, integrate, and build upon Kubeflow sub-projects within their own distributions.

> ## Scope

Reviewer 6: This looks comprehensive. I wonder if there should be clearer separation of responsibilities with other committees. Otherwise this committee will be overwhelmed.

Author: That is actually the status quo task for the platform WG or what i am doing for years.

- Try to be compatible with the popular Kubernetes clusters (Kind, Rancher, Azure Kubernetes Service, Elastic Kubernetes Service, Google Kubernetes Engine, ...).
- Provide hints and experimental examples how the consumer could integrate non-default external authentication (e.g., companies' Identity Provider) and popular non-default services on his own.
- Document the installation of Kubeflow as a platform and/or individual components including common problems and architectural overviews.

### Out of scope

- We do not enforce a specific deployment tool (e.g., ArgoCD, Flux), but we also do not want to block the usage of ArgoCD.
- The default installation shall not contain deep integration with external cloud services or closed source solutions; instead, we aim for Kubernetes-native solutions and light authentication and authorization integration with external identity providers.

### Collaboration with maintainers of kubeflow projects (e.g. KFP, Trainer, ...)

- Aid the kubeflow project maintainer in creating manifests (Helm, Kustomize) for his application.
- Aid the kubeflow project maintainer regarding security best practices.

> - Communicate with the kubeflow project maintainer regarding releases and versioning.

Reviewer 6: IMO security and releases should be separate

Author: So far in practice they are very tight together.

## Roles and Organization Management

Kubeflow's [governance model](https://github.com/kubeflow/community/blob/master/wgs/wg-governance.md) includes a plethora of different leadership roles and is quite complex and designed for larger committes. This section aims to set requirements and expectations regarding tasks, roles and promotions.

> **Subproject Owner**, **Technology Lead** or **Chair** roles were defined by trying

Reviewer 6: This should be _technical_ lead?

# Kubeflow Community Distribution

## Charter

The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable.

> ## In Scope

Reviewer: @juliusvonkohout please can you follow [other committee document format](https://github.com/kubeflow/community/blob/master/KUBEFLOW-OUTREACH-COMMITTEE.md)?

```suggestion
The Kubeflow Distribution Committee (KDC) is a vendor-neutral body dedicated to supporting a healthy ecosystem of solutions by prioritizing the development, sustainability, interoperability, and distribution of individual Kubeflow sub-projects. While the primary focus of the KDC is to empower these sub-projects as modular, high-quality components, the Kubeflow Community Distribution serves as a primary example of this effort – providing a validated community supported implementation that demonstrates the collective power of these projects in a unified environment. By maintaining this distribution, the KDC aims to lower the barrier to entry for users and provide a streamlined path to adoption, while ensuring that sub-projects remain robust and valuable.

## Charter

### In Scope (Powers)
```

### Out of Scope (Limitations)

- Anti-Ecosystem-Competitive Actions: Engaging in or permitting actions that undermine the longevity of the community or violate vendor-neutral sovereignty. This includes, but is not limited to:
      - **Proprietary Lock-in:** Mandating that sub-projects or the distribution depend on proprietary APIs or closed-source components when general, open-source alternatives exist.
       - **Core Fragmentation:** Implementing core functionality that is not upstreamed, as the KDC’s goal is to maintain a Kubernetes-native, portable, and extensible foundation.
       - **Exclusionary Integration:** Forcing dependencies that intentionally disadvantage specific vendors or environments.
       - **Note on Compatibility:** This does not exclude the development of community-supported or vendor-contributed adapters (e.g., data exporters, storage drivers, or external dependencies) designed to support specific environments, provided they remain external to the project's core functionality and do not compromise the portability of the base ecosystem.

> ### Voting

Reviewer: This should be
```suggestion
## Committee Structure

### Committee Roles

```

# Kubeflow Community Distribution committee charter

> 1. Promote the Kubeflow community’s vision, values, and mission in collaboration with the KSC, KOC, and WG Leads.

Reviewer: Could you add them under `Charter`?

E.g.

## Charter 
### In Scope (powers)
1. Promote the Kubeflow community’s vision, values, and mission in collaboration with the KSC, KOC, and WG Leads.
....
